src/plugins/cesium/interactionHelpers.ts

```diff
@@ -638,3 +640,69 @@ export function updateHeightForCartesianPositions(
       : Cartographic.toCartesian(cartographicPosition);
   });
 }
+
+export function calculateViewOnRectangle(
+  bottomLeft: Cartesian3,
+  bottomRight: Cartesian3,
+  topLeft: Cartesian3,
+  topRight: Cartesian3,
+  fovAngle = 45,
+): {
+  destination: Cartesian3;
+  orientation: DirectionUp;
+} {
+  // Calculate the midpoint of the rectangle
+  const midpoint = Cartesian3.multiplyByScalar(
+    Cartesian3.add(
+      Cartesian3.add(bottomLeft, bottomRight, new Cartesian3()),
+      Cartesian3.add(topLeft, topRight, new Cartesian3()),
+      new Cartesian3(),
+    ),
+    0.25,
+    new Cartesian3(),
+  );
+
+  const verticalExtent = Cartesian3.distance(topLeft, bottomLeft);
+  const horizontalExtent = Cartesian3.distance(bottomLeft, bottomRight);
+  const diagonalExtent = Math.sqrt(verticalExtent ** 2 + horizontalExtent ** 2);
+
+  const fov = CMath.toRadians(fovAngle);
+  const cameraDistance = diagonalExtent / (2 * Math.tan(fov / 2));
```

Comment:
Why do you use the diagonal?

Reply:
It is to fit any rectangles, wider or higher. Otherwise, we can check if the vertical or horizontal distance is bigger